src/components/ImageView/RegionView/CompassAndRulerAnnotationComponent.tsx

```diff
@@ -501,6 +515,38 @@ export const RulerAnnotation = observer((props: CompassRulerAnnotationProps) =>
                     fontFamily={region.font}
                     fontStyle={region.fontStyle}
                 />
+                <Text
```

Comment:
Perhaps we can remove the entire elements when the labels are off: `region.auxiliaryTextVisible && ...`.